### PR TITLE
Implementation of GRU

### PR DESCRIPTION
**Pull request by saromanov**
Hi, guys! First, thanks for your library, i found it nice and very useful.
I decided to implement GRU(Gated Recurrent Unit) , which descripted on the paper [Empirical Evaluation of Gated Recurrent Neural Networks on Sequence Modeling](http://arxiv.org/abs/1412.3555) and this is very similar to LSTM, but with less numbers of steps. Therefore, i took LSTM class and just "adapt" it to GRU.
And my results(training with RMSProp): after 25 iterations of GRU i've got a **loss=2.993710** and after 40 iterations of LSTM, **loss = 3.501202**.
I'm not provide example, because to run it, you can just change lstm to gru from file lstm_chime.py(from examples)

from
```python
def layer(n):
    return dict(form='bidirectional', worker='lstm', size=n)
```
to
```python
def layer(n):
    return dict(form='bidirectional', worker='gru', size=n)
```
And just run it.
What do you think about it?

### CODE REVIEW

**Comment by lmjohns3** [comment]
Fantastic, thank you!!
